datalake/backend/tests/test_processing_stack.py

```diff
@@ -0,0 +1,113 @@
+logger = logging.getLogger(__name__)
+
+
+def get_compute_environment(batch_client):
+    compute_environment_detection_response = batch_client.describe_compute_environments()
+
+    # make sure there are no more records in next pages
+    assert compute_environment_detection_response.get("nextToken", None) is None
+
+    for compute_environment in compute_environment_detection_response["computeEnvironments"]:
+        if (
+            compute_environment["tags"][ENVIRONMENT_TAG_KEY] == ENVIRONMENT_TAG_VAL
+            and compute_environment["tags"].get(APPLICATION_TAG_KEY, None) == APPLICATION_TAG_VAL
+        ):
+            datalake_compute_environment = compute_environment
+            logger.info("Datalake Compute Environment: %s", datalake_compute_environment)
+            break
+
+    return datalake_compute_environment
+
+
+def launch_compute_environment_resources(batch_client, datalake_compute_environment):
+    """
+    Trigger immediate launch of computing resources. This will significantly speed up Batch jobs
+    execution.
+    """
+    if (
+        int(datalake_compute_environment["computeResources"]["minvCpus"]) < 1
+        or int(datalake_compute_environment["computeResources"]["desiredvCpus"]) < 1
+    ):
+        batch_client.update_compute_environment(
+            computeEnvironment=datalake_compute_environment["computeEnvironmentName"],
+            computeResources={"minvCpus": 1, "desiredvCpus": 1},
+        )
+
+
+def get_state_machine(stepfunctions_client):
+    state_machines_detection_response = stepfunctions_client.list_state_machines()
+
+    # make sure there are no more records in next pages
+    assert state_machines_detection_response.get("nextToken", None) is None
+
+    for state_machine in state_machines_detection_response["stateMachines"]:
+
+        tags_detection_response = stepfunctions_client.list_tags_for_resource(
+            resourceArn=state_machine["stateMachineArn"]
+        )
+
+        # reformat returned array of tags to dictionary
+        state_machine_tags = {tag["key"]: tag["value"] for tag in tags_detection_response["tags"]}
+
+        if (
+            state_machine_tags[ENVIRONMENT_TAG_KEY] == ENVIRONMENT_TAG_VAL
+            and state_machine_tags.get(APPLICATION_TAG_KEY, None) == APPLICATION_TAG_VAL
+        ):
+            datalake_state_machine = state_machine
+            logger.info("Datalake State Machine: %s", datalake_state_machine)
+            break
+
+    return datalake_state_machine
+
+
+@mark.timeout(1200)
+@mark.infrastructure
+def test_should_successfully_run_dataset_version_creation_process(
+    batch_client, stepfunctions_client
+):
+
+    test_data_bucket = "s3://linz-geospatial-data-lake-test-data"
+    state_machine_input = json.dumps(
+        {
+            "dataset_id": "1xyz",
+            "version_id": "2xyz",
+            "type": "RASTER",
+            "metadata_url": f"{test_data_bucket}/tasman_rural_2003_1m/collection.json",
```

Comment:
If the code under test is actually going to have to read this file we should probably use an [`S3Object`](https://github.com/linz/geospatial-data-lake/pull/155#issuecomment-749300896) to populate it. That way we're not relying on external state, and the test can easily be tweaked to accommodate changes in the production code or infrastructure.

Reply:
I agree, but I can't see your S3Object in source code right now. Do you want to add it ?

Reply:
You go ahead, I've got a copy on a branch which isn't merged yet.

Reply:
S3Object used in 8e466364d2d6f8b5d86e5be609d31174704e85d0